cedar-drt/fuzz/src/lib.rs

```diff
@@ -294,6 +294,97 @@ pub fn run_val_test(
     }
 }
 
+// API for request validation
+// pub fn new<S: RequestSchema>(
+//     principal: (EntityUID, Option<Loc>),
+//     action: (EntityUID, Option<Loc>),
+//     resource: (EntityUID, Option<Loc>),
+//     context: Context,
+//     schema: Option<&S>,
+//     extensions: Extensions<'_>,
+// ) -> Result<Self, S::Error>
+
+// pub fn new_with_unknowns<S: RequestSchema>(
+//     principal: EntityUIDEntry,
+//     action: EntityUIDEntry,
+//     resource: EntityUIDEntry,
+//     context: Option<Context>,
+//     schema: Option<&S>,
+//     extensions: Extensions<'_>,
+// ) -> Result<Self, S::Error> {
+
+pub fn run_req_val_test(
+    custom_impl: &impl CedarTestImplementation,
+    schema: ValidatorSchema,
+    request: ast::Request,
+    extensions: Extensions<'_>
+) {
+    let (rust_res, rust_auth_dur) =
+        time_function(|| ast::Request::new_with_unknowns(
+            request.principal().clone(),
+            request.action().clone(),
+            request.resource().clone(),
+            request.context().cloned(),
+            Some(&schema),
+            extensions
+        ));
+    info!("{}{}", RUST_AUTH_MSG, rust_auth_dur.as_nanos());
+
+    let definitional_res = custom_impl.validate_request(&schema, &request);
+    match definitional_res {
+        TestResult::Failure(_) => {
+            ;
+        }
```

Comment:
I imagine you also want to check that Rust validation fails here? Same comment in `run_ent_val_test` below.

Reply:
I think this is not necessarily when validation fails, but more like when parsing fails or something along those lines. I wasn't sure what the desired behavior here would be and forgot to ask

Reply:
My bad, you're right. Let's panic here (and in `run_ent_val_test`) so we can quickly identify if there's an issue with the parsing.